apps/oxlint/src/main.rs

```diff
@@ -8,14 +8,19 @@ static GLOBAL: jemallocator::Jemalloc = jemallocator::Jemalloc;
 static GLOBAL: mimalloc::MiMalloc = mimalloc::MiMalloc;
 
 use oxlint::cli::{CliRunResult, LintRunner, Runner};
+use std::io::BufWriter;
 
 fn main() -> CliRunResult {
     init_tracing();
     init_miette();
 
     let command = oxlint::cli::lint_command().run();
     command.handle_threads();
-    LintRunner::new(command).run()
+    // stdio is blocked by LineWriter, use a BufWriter to reduce syscalls.
+    // See `https://github.com/rust-lang/rust/issues/60673`.
+    let mut stdout = BufWriter::new(std::io::stdout());
```

Comment:
Seems odd to have this in main, it's an implementation detail when printing to the terminal.

Reply:
Maybe have the default formatter return a `DefaultOutputFormatter::stdout` static method?

Reply:
I do not know how to capture the output for (snapshot) testing when its not coming from outside. :( 

Personally I like this idea, maybe you have some output streams which differs from the default `stdout`. 
Don't know what is possible in rust, but I could see an HTTP Stream here.